File: ruby/tictactoe/lib/tictactoe/models/player.rb
module Tictactoe
  module Models
    class Player
      attr_accessor :name, :symbol

      def initialize(name, symbol)
        raise PlayerInformationMissingError if !name || !symbol || name.empty? || symbol.empty?

        self.name = name
        self.symbol = symbol
      end
    end
  end
end

File: java/tictactoe/src/main/java/com/gagan93jtg/lld_learning/models/Player.java
package com.gagan93jtg.lld_learning.models;

import com.gagan93jtg.lld_learning.errors.PlayerInformationMissingError;

public class Player {
  String name;
  String symbol;

  public Player(String name, String symbol) throws PlayerInformationMissingError {
    if (name.isEmpty() || symbol.isEmpty()) {
      throw new PlayerInformationMissingError("Both symbol and symbol should be present");
    }

    this.name = name;
    this.symbol = symbol;
  }
}
